src/Components/Browser.JS/src/Boot.Server.ts

```diff
@@ -6,70 +6,97 @@ import { OutOfProcessRenderBatch } from './Rendering/RenderBatch/OutOfProcessRen
 import { internalFunctions as uriHelperFunctions } from './Services/UriHelper';
 import { renderBatch } from './Rendering/Renderer';
 import { fetchBootConfigAsync, loadEmbeddedResourcesAsync } from './BootCommon';
+import { CircuitHandler } from './Platform/Circuits/CircuitHandler';
 
-let connection : signalR.HubConnection;
+async function boot(reconnect: boolean = false) {
+  const circuitHandlers = new Array<CircuitHandler>();
+  window['Blazor'].addCircuitHandler = (circuitHandler: CircuitHandler) => circuitHandlers.push(circuitHandler);
 
-function boot() {
-  // In the background, start loading the boot config and any embedded resources
-  const embeddedResourcesPromise = fetchBootConfigAsync().then(bootConfig => {
-    return loadEmbeddedResourcesAsync(bootConfig);
-  });
+  await startCicuit();
 
-  connection = new signalR.HubConnectionBuilder()
-    .withUrl('_blazor')
-    .withHubProtocol(new MessagePackHubProtocol())
-    .configureLogging(signalR.LogLevel.Information)
-    .build();
+  async function startCicuit(): Promise<void> {
+    // In the background, start loading the boot config and any embedded resources
+    const embeddedResourcesPromise = fetchBootConfigAsync().then(bootConfig => {
+      return loadEmbeddedResourcesAsync(bootConfig);
+    });
+
+    const initialConnection = await initializeConnection();
+
+    // Ensure any embedded resources have been loaded before starting the app
+    await embeddedResourcesPromise;
+    const circuitId = await initialConnection.invoke<string>(
+      'StartCircuit',
+      uriHelperFunctions.getLocationHref(),
+      uriHelperFunctions.getBaseURI()
+    );
+
+    window['Blazor'].reconnect = async () => {
+      const reconnection = await initializeConnection();
+      if (!await reconnection.invoke<Boolean>('ConnectCircuit', circuitId)) {
+        throw "Failed to reconnect to the server";
```

Comment:
Typo: incorrect quotes (should be `'` to match other TypeScript style for string literals)

Reply:
Also, please use `throw new Error('Failed to reconnect to the server');`, so it captures the stack trace.